python/heterocl/schedule.py

```diff
@@ -136,9 +164,45 @@ def reuse_at(self, target, parent, axis, name=None):
             name = target.name + ".reuse"
         return self.sch.reuse_at(target, parent, axis, name)
 
+
+    def join(self, srcs, dest=None):
+        """ join multiple tensors to single dest """
+        assert len(srcs) > 0, "joined tensors should be " + \
+                "collectde from more than one srcs"
+
+        # create channels and collector stage
+        if dest is not None:
+            if isinstance(dest, tuple):
+                dest, target = dest
+                dest = self[dest]
+            elif isinstance(dest, Stage):
+                target = dest._op
+            elif isinstance(dest, tuple):
+                src, target = dest
+            else: # target tensor 
+                target = dest.tensor
+        else: target = dest
+
+        for src in srcs: 
+            if isinstance(src, tuple):
+                src, tensor = src
+                assert tensor == target, + \
+                        "inconsistent tensor joining"
+            self.sch.join(target, self[src], dest)
+
+
+    def fork(self, tensor, dests, axis=0):
+        """ fork tensor to multiple dests """
+        assert len(dests) > 0, "forked tensor should be " + \
+                "broadcast to more than one dest"
+        # dest as tvm stages
+        for dest in dests:
+            self.to(tensor, self[dest])
+
+
     def to(self, tensors, dst, src=None, axis=0,
-           stream_type=_expr.StreamExpr.Channel, 
-           depth=1, name=None):
+           stream_type=_expr.StreamExpr.FIFO, 
+           depth=1, at=None, name=None):
```

Comment:
Please update the parameter list below. What is `at`?

Reply:
`at` is used to specify the memory / storage media and port.

Reply:
I considered using `interface` but it is kinda to long. And the short `if` is not allowed.

Reply:
What does "memory/storage media & port" exactly mean?

Reply:
It means which memory the data resides in. e.g., `s.to(a, p.xcel, at=p.xcel.ddr[0])`. To move data to ddr (channel 0) attached to the accelerator.

Reply:
I think we should just use the term "media" instead of "at".
What is the meaning of "None" in this case?

Reply:
That's used for default mode. If it's not specified by users, HeteroCL will set it as DDR, and move the data to off-chip DRAM.

Reply:
Just discussed with Sean, to simplify the interface, we can use `s.to(tensor, p.xcel.hbm[0])` to move the data to accelerator, and have it saved in the HBM attached to that accelerator.